sympy/physics/mechanics/kane.py

```diff
@@ -63,9 +76,39 @@ class KanesMethod(_Methods):
     are given in the explicit form by default. In other words, the kinematic
     mass matrix is $\mathbf{k_{k\dot{q}}} = \mathbf{I}$.
     In order to get the implicit form of those matrices/vectors, you can set the
-    ``explicit_kinematics`` attribute to ``False``. So $\mathbf{k_{k\dot{q}}}$ is not
-    necessarily an identity matrix. This can provide more compact equations for
-    non-simple kinematics (see #22626).
+    ``explicit_kinematics`` attribute to ``False``. So $\mathbf{k_{k\dot{q}}}$
+    is not necessarily an identity matrix. This can provide more compact
+    equations for non-simple kinematics (see #22626).
+
+    Two linear solvers can be supplied to ``KanesMethod``: one for solving the
+    kinematic differential equations and one to solve the velocity constraints.
+    Both of these sets of equations form a linear system ``Ax = b``, which has
+    to be solved in order to obtain the equations of motion.
+
+    The implemented solvers are:
+        - ``'lu'``: LU solver which utilizes ``Matrix.LUsolve`` (default)
+        - ``'numerics'``: numerical solver which utilizes
+          ``codegen.matrix_nodes.MatrixSolve``. Sets ``.use_block_matrices`` to
+          ``True``.
+        - ``callable``: custom solver supplied by the user, which takes the
+          ``A`` matrix as the first argument and ``b`` matrix as the second.
+
+    The default solver, ``Matrix.LUsolve``, results in the fastest execution
+    time after lambdifying the solution. The weakness of this method is that it
+    can result in zero division errors (see #24780).
+
+    The numeric solver, ``codegen.matrix_nodes.MatrixSolve``, postpones the
+    solve to the numeric evaluation. When using ``lambdify`` combined with the
+    ``NumPyPrinter`` this will result in the usage of ``np.linalg.solve``. The
+    advantage of this method is that it results in numerically stable solutions.
+    It is however slower than ``Matrix.LUsolve``, at least for relatively small
+    systems. A possible disadvantage of this solver is that the resulting
+    matrices do not support indexing. The numeric solver can therefore not be
+    used for the kinematic differential equations. When used for the velocity
+    constraints, ``KanesMethod`` will automatically switch to use
+    ``BlockMatrix`` for the mass matrices and forcing vectors.
```

Comment:
I'm not sure we are ready to introduce this. I'd rather see a separate PR for this matrix solve option so we can discuss more. For example, maybe we should fix this indexing and maybe we don't use matrixsolve, but something new. Can we make this PR just to introduce a way to use different linear solvers?

Reply:
I've reverted the change.

Reply:
Thanks. I'm looking forward to getting this in, but I think more a more careful look is needed.